pkg/session/session.go

```diff
@@ -190,21 +198,25 @@ func (s *Session) keepAlive() {
 			log.Logger.Debug("session keep alive: closing keep alive")
 			return
 		case <-ticker.C:
-			readerExit := make(chan any)
-			writerExit := make(chan any)
+			if err := s.waitUntilServerReady(s.ctx); err != nil {
+				log.Logger.Errorf("session keep alive: error waiting for server ready: %v", err)
+				continue
+			}
+
 			s.closer = &closeOnce{closer: make(chan any)}
+
 			ctx, cancel := context.WithCancel(context.Background()) // create local context for each session
+
+			readerExit := make(chan any)
+			writerExit := make(chan any)
 			jar, _ := cookiejar.New(nil)
-			if err := s.checkServerHealth(ctx, jar); err != nil {
```

Comment:
Why do you change these lines? we should use jar to receive the cookie to stick the session on one of our servers...